src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useContext } from 'react';
+import { CSSTransition, TransitionGroup } from 'react-transition-group';
+
+import { TodoItem } from '../TodoItem';
+import { AuthContext } from '../../contexts/AuthContext';
+
+import { ErrorType } from '../../enums/ErrorType';
+import { Todo } from '../../types/Todo';
+import { OptionalTodo } from '../../types/OptionalTodo';
+import { OnChangeFunc } from '../../types/OnChangeFunc';
+
+type Props = {
+  todos: Todo[];
+  activeTodosNum: number;
+  tempTodoTitle: string;
+  isClearCompleted: boolean;
+  isAllToggled: boolean;
+  showError: (errorType: ErrorType) => void;
+  hideError: () => void;
+  onDeleteTodo: (todoId: number) => void;
+  onChangeTodo: OnChangeFunc;
+};
+
+export const TodoList: React.FC<Props> = React.memo(
+  ({
+    todos,
+    activeTodosNum,
+    tempTodoTitle,
+    isClearCompleted,
+    isAllToggled,
+    onDeleteTodo,
+    onChangeTodo,
+    showError,
+    hideError,
+  }) => {
+    const { id: userId = 0 } = useContext(AuthContext) || {};
+
+    const tempTodo: OptionalTodo = !tempTodoTitle
+      ? null
+      : {
+        id: 0,
+        userId,
+        title: tempTodoTitle,
+        completed: false,
+      };
```

Comment:
it looks a little weird, I suggest to transform it into function, and again handle truth case first